voidcrew/code/game/objects/items/devices/walkman/walkman.dm

```diff
@@ -0,0 +1,307 @@
+	break_sound()
+	current_song = null
+	current_listener = null
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/device/walkman/attackby(obj/item/cassette, mob/user)
+	if(istype(cassette, /obj/item/device/cassette_tape))
+		if(!tape)
+			insert_tape(cassette)
+			playsound(src,'sound/weapons/handcuffs.ogg',20,1)
+			to_chat(user,("You insert \the [cassette] into \the [src]"))
+		else
+			to_chat(user,("Remove the other tape first!"))
+
+/obj/item/device/walkman/attack_self(mob/user)
+	..()
+
+	if(!current_listener)
+		current_listener = user
+		START_PROCESSING(SSobj, src)
+	if(istype(tape))
+		if(paused)
+			play()
+			to_chat(user,("You press [src]'s 'play' button"))
+		else
+			pause()
+			to_chat(user,("You pause [src]"))
+		update_icon()
+	else
+		to_chat(user,("There's no tape to play"))
+	playsound(src,'sound/machines/click.ogg',20,1)
+
+/obj/item/device/walkman/AltClick(mob/user)
+	if(tape)
+		eject_tape(user)
+		return
+	else
+		..()
+
+/obj/item/device/walkman/CtrlClick(mob/user)
+	if(tape)
+		next_song(user)
+		return
+
+///This is called when sound needs to be broken ie you die or lose access to it
+/obj/item/device/walkman/proc/break_sound()
+	var/sound/break_sound = sound(null, 0, 0, SOUND_CHANNEL_WALKMAN)
+	break_sound.priority = 255
+	update_song(break_sound, current_listener, 0)
+
+/*Called when songs are updated ie volume change
+ *Arguements: mob/user -> the current user of the walkman
+ * sound/noise -> the sound that is being directed to the user
+ */
+/obj/item/device/walkman/proc/update_song(sound/noise, mob/user, flags = SOUND_UPDATE)
+	if(!istype(user) || !istype(noise)) return
+	if(HAS_TRAIT(user, TRAIT_DEAF))
+		flags |= SOUND_MUTE
+	noise.status = flags
+	noise.volume = src.volume
+	noise.channel = SOUND_CHANNEL_WALKMAN
+	sound_to(user,noise)
+
+/*Called when music is paused by the user
+ *Arguements: mob/user -> the current user of the walkman
+ */
+/obj/item/device/walkman/proc/pause(mob/user)
+	if(!current_song) return
+	paused = TRUE
+	update_song(current_song,current_listener, SOUND_PAUSED | SOUND_UPDATE)
+
+///Handles the actual playing of the sound to the current_listener
+/obj/item/device/walkman/proc/play()
+	if(!current_song)
+		if(current_playlist.len > 0)
+			current_song = sound(current_playlist[pl_index], 0, 0, SOUND_CHANNEL_WALKMAN, volume)
+			current_song.status = SOUND_STREAM
+		else
+			return
+	paused = FALSE
+	if(current_song.status & SOUND_PAUSED)
+		update_song(current_song,current_listener)
+	else
+		update_song(current_song,current_listener,0)
+
+	update_song(current_song,current_listener)
+
+/*Called when
+ *Arguements: obj/item/device/cassette_tape/CT -> the cassette in question that you are inserting into the walkman
+ */
+/obj/item/device/walkman/proc/insert_tape(obj/item/device/cassette_tape/CTape)
+	if(tape || !istype(CTape)) return
+
+	tape = CTape
+	CTape.forceMove(src)
+
+	update_icon()
+	paused = TRUE
+	pl_index = 1
+	if(tape.songs["side1"] && tape.songs["side2"])
```

Comment:
early return

Reply:
isn't the lines  a few above an early return?

Reply:
Yes but for a different case.. also it doesn't really matter here, so its fine